src/pip/_internal/commands/show.py

```diff
@@ -138,9 +138,10 @@ def print_results(distributions, list_files=False, verbose=False):
             write_output("---")
 
         name = dist.get('name', '')
+        canonical_name = canonicalize_name(name)
         required_by = [
             pkg.project_name for pkg in pkg_resources.working_set
-            if name in [required.name for required in pkg.requires()]
+            if canonical_name in [required.name for required in pkg.requires()]
```

Comment:
I think we should probably be doing `canonicalize_name(required.name)` here, that may fix the issue you mentioned with the "Required-by" of "Req-Cap-Todo" missing "simpleone" in our new test.

Reply:
You are indeed correct, I really need to setup up my pycharm debugger with tox.
Everything passes now.

I also think we should extract this list comprehension to a method, under the rule of "a function should do one thing and one thing only".
I would've done so in a personal project (without thinking too much about it) but since I'm new here I'm checking with you.

Reply:
I agree this could be improved, and moving it into `search_packages_info` seems like it would work and make more sense.

If you make the change in a separate commit then it should be OK here. It may take a little longer for someone else to review and merge, or maybe not - everyone is a volunteer so it can vary wildly. A few things that could help:

1. If you squash/rewrite your commits into meaningful "chunks" of changes that have clear diffs (maybe one commit for the primary change here and then one for moving the `requires_by` calculation into `search_packages_info`) that would make it easier to evaluate this PR
2. Give each of those commits a clear message (reduces noise compared to having a handful of commits)

These aren't required by any means, but can reduce the work required to review and can be useful to know how to do.

In general if the primary change being made is larger or more complex I would probably do refactoring in separate PRs before the "main" one if I think they are required to make the "main" changes clear. If the refactoring is more like a nice-to-have but doesn't really improve understanding the "main" change then I would probably do it in a followup PR after the "main" one. You don't have to figure this out ahead of time on your own - if some refactoring is needed then it's OK to put the "main" PR on hold, do refactoring in a separate PR, and then rebase and continue with the "main" one when the refactoring is merged. Similarly it's OK to say in a PR comment or issue that some refactoring followup is needed.

Reply:
Thank you very much for all the feedback, I appreciate it greatly.
I've been working in a python team for about 3 years but open source feedback adds much more.

Just one more question before I push:
I have an option to add a private:
def _get_requiring_packages(package_name):
```
    return [
        pkg.project_name for pkg in pkg_resources.working_set
        if canonicalize_name(package_name) in
           [canonicalize_name(required.name) for required in pkg.requires()]
    ]
```
or to do it directly in `search_packages_info`:
```
 package = {
            'name': dist.project_name,
            'version': dist.version,
            'location': dist.location,
            'requires': [dep.project_name for dep in dist.requires()],
            'required_by': [
                pkg.project_name for pkg in pkg_resources.working_set
                if canonicalize_name(dist.project_name) in
                [canonicalize_name(required.name) for required in pkg.requires()]
            ]
        }

```
I feel like the first option is better (and passes the 79 char per line limit!) but again, wanted to make sure before I push.

Reply:
1 seems better to me, but I would make it a free function in `search_packages_info` and drop the leading underscore (because its scope will already prevent it from being used outside, marking it private would be redundant IMO).